src/applyMiddleware.js

```diff
@@ -24,7 +24,7 @@ export default function applyMiddleware(...middlewares) {
 
     var middlewareAPI = {
       getState: store.getState,
-      dispatch: (action) => dispatch(action)
+      dispatch: (action) => store.dispatch(action)
```

Comment:
Yes you are right. There still room for little improvements.  I started discussion at #1446 you can share your view there